code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -989,8 +989,16 @@ pub mod pallet {
 					false, // pallet account, doesn't need to be kept alive
 				)?;
 			}
-			// burn the shares
-			T::Assets::burn_from(share_asset_id, &fnft_asset_account, stake.share)?;
+			// transfer the shares to pool account (not burning to avoid affecting pool share
+			// calculation(share/total_issuance) for other stakes)
+			T::Assets::transfer(
```

Comment:
Are we sure this is a matter of burning or not instead of a matter of when we burn the share asset?
With this approach, we seem to be doing a mint only share asset.

Reply:
The share asset burning will affect existing position claims whenever we burn shares as the ratios are affected, so we need to keep minting and not burning as it stands. This is also according to the original pool math that assumes total_shares to be immutable.